iHome/ihome.frontend/src/app/pages/rooms/device/rgbcontrol/rgbcontrol.component.ts

```diff
@@ -59,9 +61,16 @@ export class RGBControlComponent implements OnInit {
     this.updateDevice();
   }
 
+  updateMode(mode: number){
+    this.data = {
+      ...this.data,
+      Mode: mode
+    };
+    this.updateDevice();
+  }
+
   private updateDevice(){
-    console.log(this.data);
-    this.api.setDeviceData(this.device.deviceId, JSON.stringify(this.data)).subscribe();
+    this.api.setDeviceData(this.device.id, JSON.stringify(this.data)).subscribe();
```

Comment:
Tak właściwie to po co my tutaj subskrybujemy? To nie tak że nie powinniśmy, tylko nie pamiętam do końca jak eventy/async działał w angularze

Reply:
Jak robisz asynca to musisz dać subscribe bo nie wykona requesta